### PR TITLE
Rename RPC methods with wifi prefix.

### PR DESCRIPTION
**Pull request by kaishi05**
This change is [Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/209)

### CODE REVIEW

**Review by xpconanfan — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/209)* status: 1 of 2 files reviewed, 1 unresolved discussion (waiting on @kaishi05)

___
*[`src/main/java/com/google/android/mobly/snippet/bundled/WifiAwareManagerSnippet.java` line 47 at r1](https://reviewable.io/reviews/google/mobly-bundled-snippets/209#-O4qL8se4LKPhyimR3mt:-O4qL8se4LKPhyimR3mu:b-e9hrso) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/d3d513b919c07b321d9106a93e429c48dc6ded1b/src/main/java/com/google/android/mobly/snippet/bundled/WifiAwareManagerSnippet.java#L47)):*
> ```Java
>   public boolean wifiAwareIsAvailable() throws Throwable {
>     if (!mIsAwareSupported || mWifiAwareManager == null) {
>       return false;
> ```

we should throw an exception here instead.

Is available is clearly a different concept from `is supported`.

When it's not supported, `is available` is an invalid question, hence it should raise?

**Comment by kaishi05** [comment]
As suggested, throw an exception when it is not supported.

File: src/main/java/com/google/android/mobly/snippet/bundled/WifiAwareManagerSnippet.java [hunk 1]
```diff
@@ -26,26 +26,34 @@
...
+        if (mIsAwareSupported) {
+            mWifiAwareManager = (WifiAwareManager) mContext.getSystemService(Context.WIFI_AWARE_SERVICE);
+        }
+        Utils.adaptShellPermissionIfRequired(mContext);
+    }
+
+    /** Checks if Aware is available. This could return false if WiFi or location is disabled. */
+    @Rpc(description = "check if Aware is available.")
+    public boolean wifiAwareIsAvailable() throws WifiAwareManagerSnippetException {
+        if (!mIsAwareSupported || mWifiAwareManager == null) {
```
Comment (xpconanfan) [inline src/main/java/com/google/android/mobly/snippet/bundled/WifiAwareManagerSnippet.java:53]: one of these two is sufficient for determining whether wifi aware is supported on this device right?
can we simplify the logic and only have one signal var instead?

Comment (kaishi05) [reply src/main/java/com/google/android/mobly/snippet/bundled/WifiAwareManagerSnippet.java]: Done.

File: src/main/java/com/google/android/mobly/snippet/bundled/WifiAwareManagerSnippet.java [hunk 2]
```diff
@@ -50,7 +49,7 @@ public WifiAwareManagerSnippet() throws Throwable {
     /** Checks if Aware is available. This could return false if WiFi or location is disabled. */
     @Rpc(description = "check if Aware is available.")
     public boolean wifiAwareIsAvailable() throws WifiAwareManagerSnippetException {
-        if (!mIsAwareSupported || mWifiAwareManager == null) {
+        if (mWifiAwareManager == null) {
```
Comment (xpconanfan) [inline src/main/java/com/google/android/mobly/snippet/bundled/WifiAwareManagerSnippet.java:52]: let's use `isAwareSupported` instead as it's more direct?

Comment (kaishi05) [reply src/main/java/com/google/android/mobly/snippet/bundled/WifiAwareManagerSnippet.java:52]: There is a small chance that mWifiAwareManager could be still null even if isAwareSupported = true. So I prefer the null check.

Comment (xpconanfan) [reply src/main/java/com/google/android/mobly/snippet/bundled/WifiAwareManagerSnippet.java:52]: But that null doesn't necessarily mean wifi aware is not supported.
If you are worried about that edge case, we should handle that separately with a different exception

Comment (kaishi05) [reply src/main/java/com/google/android/mobly/snippet/bundled/WifiAwareManagerSnippet.java:52]: Null means WifiAware service is not running and as the result Aware is not supported.

Comment (xpconanfan) [reply src/main/java/com/google/android/mobly/snippet/bundled/WifiAwareManagerSnippet.java:52]: "supported" means this device can support aware.
"available" means this device can perform wifi aware actions right now.

the edge case you brought up means Aware is supported on this device, but wifi aware manager being null
but that doesn't mean the wifi aware manager can't be null on this device ever?

Comment (kaishi05) [reply src/main/java/com/google/android/mobly/snippet/bundled/WifiAwareManagerSnippet.java:52]: Okay. If you were more clear on original suggestion, we would have less back and forth on this.  I made the change. Let's move on.

**Review by xpconanfan — APPROVED** [review]

**Pull request merged**